finetuner/tuner/callback/model_checkpoint.py

```diff
@@ -0,0 +1,141 @@
+    `ModelCheckepointCallback` is used in conjunction with training
+    using `finetuner.fit()`
+
+    A few options this callback provides include:
+    - Whether to only keep the model that has achieved the "best performance" so
+        far, or whether to save the model at the end of every epoch regardless of
+        performance.
+    - Definition of 'best'; which quantity to monitor and whether it should be
+        maximized or minimized.
+    - The frequency it should save at. Currently, the callback supports saving at
+        the end of every epoch, or after a fixed number of training batches.
+    """
+
+    def __init__(
+        self,
+        filepath: str = None,
+        save_best_only: Optional[bool] = False,
+        monitor: Optional[str] = 'val_loss',
+        mode: Optional[str] = 'auto',
+    ):
+        """
+        :param filepath: string or `PathLike`, path to save the model file.
+        :param save_best_only: if `save_best_only=True` only the best model
+        model will be saved according to the quantity monitored
+        :param monitor: if `monitor='loss'` best bodel saved will be according
+        to the training loss, if `monitor='val_loss'` best model saved will be
+        according to the validation loss
+        :param mode: one of {'auto', 'min', 'max'}. If `save_best_only=True`, the
+        decision to overwrite the current save file is made based on either
+        the maximization or the minimization of the monitored quantity.
+        For `val_acc`, this should be `max`, for `val_loss` this should be
+        `min`, etc. In `auto` mode, the mode is set to `max` if the quantities
+        monitored are 'acc' or start with 'fmeasure' and are set to `min` for
+        the rest of the quantities.
+        """
+        self.__logger = JinaLogger(self.__class__.__name__)
+        self.__filepath = filepath
+        self.__save_best_only = save_best_only
+        self.__monitor = monitor
+        if not filepath:
+            raise ValueError(
+                '``filepath`` parameter is mandatory. Pass it in parameters'
+            )
+
+        if mode not in ['auto', 'min', 'max']:
+            self.__logger.warning(
+                'ModelCheckpoint mode %s is unknown, ' 'fallback to auto mode.', mode
+            )
+            mode = 'auto'
+
+        if mode == 'min':
+            self.__monitor_op = np.less
+            self.__best = np.Inf
+        elif mode == 'max':
+            self.__monitor_op = np.greater
+            self.__best = -np.Inf
+        else:
+            if 'acc' in self.__monitor or self.__monitor.startswith('fmeasure'):
+                self.__monitor_op = np.greater
+                self.__best = -np.Inf
+            else:
+                self.__monitor_op = np.less
+                self.__best = np.Inf
+
+    def get_monitor_op(self):
+        return self.__monitor_op
+
+    def get_best(self):
+        return self.__best
```

Comment:
These are unnecessary

Reply:
Used in tests to check if mode functions correctly

Reply:
You can access the private attributes directly in test, nothing wrong with that I would say

Reply:
okay I'll remove them